spec/system/user_page/staff_info_spec.rb

```diff
@@ -17,4 +17,43 @@
       expect(user_page).to have_warning_messages_path(user)
     end
   end
+
+  context "for flagged posts" do
+    before do
+      %i[disagree ignore delete_and_ignore].each do |review_action|
+        PostActionCreator
+          .off_topic(admin, Fabricate(:post, user: user))
+          .reviewable
+          .perform(admin, review_action)
+      end
+    end
+
+    context "when there are no approved flagged posts" do
+      it "should not display a flagged-posts staff counter" do
+        user_page.visit(user)
+        expect(user_page).to have_no_staff_info_flagged_posts_counter
+      end
+    end
+
+    context "when there are approved flagged posts" do
+      before do
+        2.times do
+          PostActionCreator
+            .off_topic(admin, Fabricate(:post, user: user))
+            .reviewable
+            .perform(admin, :agree_and_keep)
+        end
+      end
+
+      it "should show the right count in the flagged-posts staff counter" do
+        user_page.visit(user)
+        expect(user_page).to have_staff_info_flagged_posts_count(count: 2)
+      end
+
+      it "should have the right link to user's flagged posts in the flagged-posts staff counter" do
+        user_page.visit(user).staff_info_flagged_posts_counter.click
+        expect(user_page).to have_reviewable_flagged_posts_path(user)
+      end
```

Comment:
```suggestion
      it "should show display a link to user's flagged posts with the right count in the flagged-posts staff counter" do
        user_page.visit(user)

        expect(user_page).to have_staff_info_flagged_posts_count(count: 2)

        user_page.visit(user).staff_info_flagged_posts_counter.click

        expect(user_page).to have_reviewable_flagged_posts_path(user)
      end
```

Minor one but since system tests are more expensive to run, we can combine the two assertions into a single test group.